### PR TITLE
feat(bigquery): Add example for creating a table with tag attached

### PR DESCRIPTION
**Pull request by pratikmore0290**
## Description

Fixes #353964666

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [x] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [x] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s): https://cloud.google.com/bigquery/docs/tags#attach_tags_when_you_create_a_new_table

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

File: bigquery/bigquery_create_table_attach_tag/main.tf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  description         = "table description"
+  deletion_protection = false # set to "true" in production
+
+  # Attach tags to the table
+  resource_tags = {
+    "${google_tags_tag_key.env_tag_key.namespaced_name}" : "${google_tags_tag_value.env_tag_value.short_name}",
+    "${google_tags_tag_key.department_tag_key.namespaced_name}" : "${google_tags_tag_value.department_tag_value.short_name}"
+  }
+}
+# [END bigquery_create_table_attach_tag]
```
Comment (glasnt) [inline bigquery/bigquery_create_table_attach_tag/main.tf:65]:
```suggestion
# [END bigquery_create_table_attach_tag]

```

File: bigquery/bigquery_create_table_attach_tag/main.tf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+resource "google_bigquery_table" "default" {
+  dataset_id          = google_bigquery_dataset.default.dataset_id
+  table_id            = "mytable"
+  description         = "table description"
+  deletion_protection = false # set to "true" in production
+
+  # Attach tags to the table
+  resource_tags = {
+    "${google_tags_tag_key.env_tag_key.namespaced_name}" : "${google_tags_tag_value.env_tag_value.short_name}",
+    "${google_tags_tag_key.department_tag_key.namespaced_name}" : "${google_tags_tag_value.department_tag_value.short_name}"
```
Comment (glasnt) [inline bigquery/bigquery_create_table_attach_tag/main.tf:62]: You don't need to use string interpolation for single variables, but TIL you still have to account for ambiguous attribute keys. This should be valid: 
```suggestion
    (google_tags_tag_key.env_tag_key.namespaced_name): google_tags_tag_value.env_tag_value.short_name,
    (google_tags_tag_key.department_tag_key.namespaced_name): google_tags_tag_value.department_tag_value.short_name
```

File: bigquery/bigquery_create_table_attach_tag/main.tf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+resource "google_bigquery_table" "default" {
+  dataset_id          = google_bigquery_dataset.default.dataset_id
+  table_id            = "mytable"
+  description         = "table description"
+  deletion_protection = false # set to "true" in production
+
+  # Attach tags to the table
+  resource_tags = {
+    (google_tags_tag_key.env_tag_key.namespaced_name): google_tags_tag_value.env_tag_value.short_name,
+    (google_tags_tag_key.department_tag_key.namespaced_name): google_tags_tag_value.department_tag_value.short_name
```
Comment (glasnt) [inline bigquery/bigquery_create_table_attach_tag/main.tf:62]: Turns out this needs a space
```suggestion
    (google_tags_tag_key.env_tag_key.namespaced_name) : google_tags_tag_value.env_tag_value.short_name,
    (google_tags_tag_key.department_tag_key.namespaced_name) : google_tags_tag_value.department_tag_value.short_name
```

**Review by glasnt — APPROVED** [review]

**Review by wj-chen — APPROVED** [review]

**Comment by glasnt** [comment]
Terraform destroy is failing due to dependency issues: 

```
Error: Error waiting for Deleting TagValue: Error code 9, message: Cannot delete tag value, tagValues/[value], because it is still attached to resources in 'us' region. To delete this tag value, remove all holds and then remove it from all resources. At least one binding was found to an active or deleted resource in the 'us' region.
```

**Comment by glasnt** [comment]
Logged upstream issue

**Comment by priyanshsaxena** [comment]
The resource is adding tags at creation. Since Terraform can deduce the implicit dependency of the table-resource on tag-value-resource because of the resource_tags field, the creation-order should be value-then-table. [Terraform documentation](https://developer.hashicorp.com/terraform/tutorials/configuration-language/dependencies#destroy-resources) mentions that both implicit and explicit dependencies should affect the order of creation and destruction.

One way to explicitly mark the dependency is to use the `depends_on` meta-field. Please try to run the `graph` command before and after adding the explicit dependency to see if changes anything.

```
terraform graph -type=plan | dot -Tpng >graph.png
```

**Comment by glasnt** [comment]
> The resource is adding tags at creation. Since Terraform can deduce the implicit dependency of the table-resource on tag-value-resource because of the resource_tags field, the creation-order should be value-then-table. [Terraform documentation](https://developer.hashicorp.com/terraform/tutorials/configuration-language/dependencies#destroy-resources) mentions that both implicit and explicit dependencies should affect the order of creation and destruction.
> 
> One way to explicitly mark the dependency is to use the `depends_on` meta-field. Please try to run the `graph` command before and after adding the explicit dependency to see if changes anything.
> 
> ```
> terraform graph -type=plan | dot -Tpng >graph.png
> ```

I've tried adding `depends_on` in multiple forms in local testing, but it didn't work, hence the upstream report.

**Comment by glasnt** [comment]
To unblock this for now, we can add `test.yaml` to remove testing this sample.

**Pull request merged**